frontend/src/components/JsonViewer/json-box.vue

```diff
@@ -0,0 +1,160 @@
+<script>
+import JsonString from "./types/json-string.vue";
+import JsonUndefined from "./types/json-undefined.vue";
+import JsonNumber from "./types/json-number.vue";
+import JsonBoolean from "./types/json-boolean.vue";
+import JsonObject from "./types/json-object.vue";
+import JsonArray from "./types/json-array.vue";
+import JsonFunction from "./types/json-function.vue";
+import JsonDate from "./types/json-date.vue";
+import JsonRegexp from "./types/json-regexp.vue";
+import { h } from "vue";
+export default {
+  name: "JsonBox",
+  inject: ["expandDepth", "keyClick"],
+  props: {
+    value: {
+      type: [Object, Array, String, Number, Boolean, Function, Date],
+      default: null,
+    },
+    keyName: {
+      type: String,
+      default: "",
+    },
+    sort: Boolean,
+    depth: {
+      type: Number,
+      default: 0,
+    },
+    previewMode: Boolean,
+  },
+  data() {
+    return {
+      expand: false,
+    };
+  },
+  mounted() {
+    // this.expand = this.previewMode || (this.depth >= this.expandDepth ? false : true);
+  },
+  methods: {
+    toggle() {
+      this.expand = !this.expand;
+
+      try {
+        this.$el.dispatchEvent(new Event("resized"));
+      } catch (e) {
+        // handle IE not supporting Event constructor
+        var evt = document.createEvent("Event");
+        evt.initEvent("resized", true, false);
+        this.$el.dispatchEvent(evt);
+      }
+    },
+  },
+  render() {
+    let elements = [];
+    let dataType;
+
+    if (this.value === null || this.value === undefined) {
```

Comment:
@bradleySuira this can be improved by having a mapping object with keys equals to the this.value and values equals to the dataTypes.

Reply:
tried but in this case we are not using always `typeof` and we have some other cases for complex types like `Array` and `Dates`